ethereum/p2p/src/main/java/tech/pegasys/pantheon/ethereum/p2p/permissioning/NodeWhitelistController.java

```diff
@@ -180,6 +191,29 @@ public boolean isPermitted(final Peer node) {
     return nodesWhitelist;
   }
 
+  public synchronized void reload() throws RuntimeException {
+    final List<Peer> currentAccountsList = new ArrayList<>(nodesWhitelist);
+    nodesWhitelist.clear();
+
+    try {
+      final PermissioningConfiguration updatedConfig =
+          PermissioningConfigurationBuilder.permissioningConfigurationFromToml(
+              configuration.getConfigurationFilePath(),
+              configuration.isNodeWhitelistEnabled(),
+              configuration.isAccountWhitelistEnabled());
+
+      readNodesFromConfig(updatedConfig);
+      this.configuration = updatedConfig;
```

Comment:
It's probably an unlikely case, but should the config be reverted if the Exception is thrown?

Reply:
We only update the config as the last line inside the try-catch block. Any exception thrown before that will result in the configuration object not being updated.